core/security/src/main/java/org/apache/isis/core/security/authentication/Authentication.java

```diff
@@ -91,8 +101,24 @@ default Type getType() {
     // -- WITHERS
 
     /**
-     * Returns a copy with given {@code executionContext}.
-     * @param executionContext
+     * Returns a copy with given {@code interactionContext}.
+     * @param interactionContext
      */
-    Authentication withExecutionContext(ExecutionContext executionContext);
+    Authentication withInteractionContext(InteractionContext interactionContext);
+
+    /**
+     * Looks up an {@link Authentication} from the provided {@link InteractionContext}.
+     *
+     * <p>
+     *     If the {@link InteractionContext} was created as the result of an authentication process (as opposed to
+     *     programmatically) then there will be an attached {@link Authentication} object.
+     * </p>
+     *
+     * @param interactionContext
+     * @return
+     */
+    public static Optional<Authentication> authenticationFrom(final @NonNull InteractionContext interactionContext) {
```

Comment:
relates to a comment above, that InteractionContext should be immutable; I hope there are alternative ways to get the (current) Authentication; not sure

Reply:
per comment above, perhaps use InteractionLayer ?